storage/client/internal/parse_rfc3339.cc

```diff
@@ -0,0 +1,181 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "storage/client/internal/parse_rfc3339.h"
+#include "google/cloud/internal/throw_delegate.h"
+#include <cstdio>
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+
+namespace {
+[[noreturn]] void ReportError(std::string const& timestamp, char const* msg) {
+  std::ostringstream os;
+  os << "Error parsing RFC 3339 timestamp: " << msg
+     << " Valid format is YYYY-MM-DD[Tt]HH:MM:SS[.s+](Z|+HH:MM), got="
+     << timestamp;
+  google::cloud::internal::RaiseInvalidArgument(os.str());
+}
+
+std::chrono::system_clock::time_point ParseDateTime(
+    char const*& buffer, std::string const& timestamp) {
+  // Use std::mktime to compute the number of seconds because RFC 3339 requires
+  // working with civil time, including the annoying leap seconds, and mktime
+  // does.
+  int year, mon, mday;
+  char date_time_separator;
+  int hh, mm, ss;
```

Comment:
s/hh/hours/ ?
s/mm/minutes/ ?
s/ss/seconds/ ?

Just makes it much easier to read.

Reply:
Fixed.